frameworks/kafka/docs/limitations.md

```diff
@@ -17,6 +17,17 @@ The "disk" configuration value is denominated in MB. We recommend you set the co
 
 When Kerberos is enabled, the broker VIP is disabled as Kerberized clients will not be able to use it. This is because each Kafka broker uses a specific Kerberos principal and cannot accept connections from a single unified principal which the VIP would require.
 
+### Toggling Kerberos
+
+Kerberos authentication can be toggled (enabled / disabled), but this triggers a rolling restart of the cluster. Clients configured with the old security settings will lose connectivity during and after this process. It is recommended that backups are made and downtime is scheduled.
+
+### Toggling TLS
+
+Transport encryption using TLS can be toggled (enabled / disabled), but will trigger a rolling restart of the cluster. As each broker restarts, a client may lose connectivity based on its security settings and the value of the `service.security.transport_encryption.allow_plaintext` configuration option. It is recommended that backups are made and downtime is scheduled.
+
+In order to enable TLS, a service account and corresponding secret is required. Since it is not possible to change the service account used by a service, it is recommended that the service is deployed with an explicit service account to allow for TLS to be enabled at a later stage.
```

Comment:
> Since it is not possible to change the service account used by a service

Is this documented somewhere? I just tried it with Elastic and this is what `update status` shows:
```
$ dcos beta-elastic --name=beta-elastic update status
deploy (serial strategy) (ERROR)
├─ master-update (serial strategy) (COMPLETE)
│  ├─ master-0:[node] (COMPLETE)
│  ├─ master-1:[node] (COMPLETE)
│  └─ master-2:[node] (COMPLETE)
├─ data-update (serial strategy) (COMPLETE)
│  ├─ data-0:[node] (COMPLETE)
│  └─ data-1:[node] (COMPLETE)
├─ ingest-update (serial strategy) (COMPLETE)
└─ coordinator-update (serial strategy) (COMPLETE)
   └─ coordinator-0:[node] (COMPLETE)

Errors:
- Field: 'TaskVolumes[taskname:node]'; Transition: '[name: disk, value: type: SCALAR scalar { value: 10000.0 }, role: beta-elastic-role, pre-reserved-role: *, principal: elastic, type: 'ROOT', container-path: 'container-path']' => '[name: disk, value: type: SCALAR scalar { value: 10000.0 }, role: beta-elastic-role, pre-reserved-role: *, principal: elastic-same-key, type: 'ROOT', container-path: 'container-path']'; Message: 'Volumes must be equal.'
- Field: 'TaskVolumes[taskname:node]'; Transition: '[name: disk, value: type: SCALAR scalar { value: 2000.0 }, role: beta-elastic-role, pre-reserved-role: *, principal: elastic, type: 'ROOT', container-path: 'container-path']' => '[name: disk, value: type: SCALAR scalar { value: 2000.0 }, role: beta-elastic-role, pre-reserved-role: *, principal: elastic-same-key, type: 'ROOT', container-path: 'container-path']'; Message: 'Volumes must be equal.'
- Field: 'TaskVolumes[taskname:node]'; Transition: '[name: disk, value: type: SCALAR scalar { value: 2000.0 }, role: beta-elastic-role, pre-reserved-role: *, principal: elastic, type: 'ROOT', container-path: 'container-path']' => '[name: disk, value: type: SCALAR scalar { value: 2000.0 }, role: beta-elastic-role, pre-reserved-role: *, principal: elastic-same-key, type: 'ROOT', container-path: 'container-path']'; Message: 'Volumes must be equal.'
- Field: 'TaskVolumes[taskname:node]'; Transition: '[name: disk, value: type: SCALAR scalar { value: 1000.0 }, role: beta-elastic-role, pre-reserved-role: *, principal: elastic, type: 'ROOT', container-path: 'container-path']' => '[name: disk, value: type: SCALAR scalar { value: 1000.0 }, role: beta-elastic-role, pre-reserved-role: *, principal: elastic-same-key, type: 'ROOT', container-path: 'container-path']'; Message: 'Volumes must be equal.'
```

Reply:
I'm not sure whether we have documentation of the limitation (@benclarkwood @gabrielhartmann?).

Have you tried using the `elastic` service account that is automatically generated and just adding a secret to that?

Reply:
Yes the default service account works, what doesn't work is the thing you mentioned: changing it (i.e. `elastic` -> `elastic-same-key`), even though the new one has the same permissions and secret.

I asked mostly because I didn't know of that limitation and couldn't find it in the documentation, and also because the error message shown is kinda cryptic. Maybe I'm unaware of something other than `update status` that shows a better error message...